autoemulate/experimental/emulators/gaussian_process/exact.py

```diff
@@ -133,6 +140,13 @@ def __init__(  # noqa: PLR0913 allow too many arguments since all currently requ
         self.epochs = epochs
         self.lr = lr
         self.activation = activation
+        self.optimizer = self.optimizer_cls(self.parameters(), lr=self.lr)  # type: ignore[call-arg] since all optimizers include lr
+        # Extract scheduler-specific kwargs if present
+        scheduler_kwargs = kwargs.pop("scheduler_kwargs", {})
+        if self.scheduler_cls is None:
+            self.scheduler = None
+        else:
+            self.scheduler = self.scheduler_cls(self.optimizer, **scheduler_kwargs)
```

Comment:
Perhaps it would be worth refactoring this into a function or method on a base class (e.g. `construct_scheduler`) as repeated a few times? An alternative to having to init the other variables too like the optimizer etc would be good but I think that would make sense in a new issue/PR.